core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -76,6 +76,9 @@ final class KafkaMetadataLog private (
     if (records.sizeInBytes == 0)
       throw new IllegalArgumentException("Attempt to append an empty record set")
 
+    if (epoch < lastFetchedEpoch())
```

Comment:
Do we need sth even more strict such that the diff between append epoch and last epoch are within 1? In what case will epoch + 2 be valid?